src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -12,19 +22,185 @@ namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+                    {
+                        var operation = GetOperationWalkingUpParentChain(referenceNode, semanticModel);
+                        SyntaxNode nodeToReplaceOpt = null;
+                        switch (operation)
+                        {
+                            case IMemberReferenceOperation memberReference:
+                                if (IsReplacableOperation(memberReference.Instance))
+                                {
+                                    nodeToReplaceOpt = GetSyntaxNodeToReplace(memberReference);
+                                }
+
+                                break;
 
+                            case IInvocationOperation invocation:
+                                if (IsReplacableOperation(invocation.Instance))
+                                {
+                                    nodeToReplaceOpt = GetExpressionOfInvocation(invocation.Syntax);
+                                }
+
+                                break;
+                        }
+
+                        if (nodeToReplaceOpt != null)
+                        {
+                            // Fetch the symbol for the node to replace - note that this might be
+                            // different from the original symbol due to generic type arguments.
+                            var symbolInfo = semanticModel.GetSymbolInfo(nodeToReplaceOpt, cancellationToken);
+                            if (symbolInfo.Symbol == null)
+                            {
+                                continue;
+                            }
+
+                            SyntaxNode memberName;
+                            var typeArgumentsOpt = GetTypeArguments(referenceNode);
+                            memberName = typeArgumentsOpt != null ?
+                                editor.Generator.GenericName(symbolInfo.Symbol.Name, typeArgumentsOpt) :
+                                editor.Generator.IdentifierName(symbolInfo.Symbol.Name);
+
+                            var newNode = editor.Generator.MemberAccessExpression(
+                                    expression: editor.Generator.TypeExpression(symbolInfo.Symbol.ContainingType),
+                                    memberName: memberName)
+                                .WithLeadingTrivia(nodeToReplaceOpt.GetLeadingTrivia())
+                                .WithTrailingTrivia(nodeToReplaceOpt.GetTrailingTrivia())
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            editor.ReplaceNode(nodeToReplaceOpt, newNode);
+                        }
+                    }
+                }
+
+                document = document.WithSyntaxRoot(editor.GetChangedRoot());
+                solution = document.Project.Solution;
+            }
+
+            return solution;
+
+            // Local functions.
+            IOperation GetOperationWalkingUpParentChain(SyntaxNode node, SemanticModel semanticModel)
```

Comment:
This feels like an useful extension method for semantic model

Reply:
Let me refactor, thanks